lib/xboxrt/libc_extensions/direct.c

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: MIT
+
+// SPDX-FileCopyrightText: 2023 ExoSkye
+
+// Part of Microsoft CRT
+
+#include <errno.h>
+#include <string.h>
+#include <stdbool.h>
+#include <windows.h>
+
+// Made referencing https://www.digitalmars.com/rtl/direct.html, retrieved 2023-06-10, copyrighted 1999-2018 by Digital Mars
+
+
+/*
+ * TODO: Make a working directory system?
+ */
+
+int _chdir(char* path) {
+    return 0;
+}
+
+int _chdrive(int drive) {
+    return 0;
+}
+
+char* _getcwd(char* buffer, size_t length) {
+    errno = -EINVAL;
+    return NULL;
+}
+
+char* _getwd(char* path_name) {
+    errno = -EINVAL;
+    return NULL;
+}
+
+int _getdrive(void) {
+    errno = -EINVAL;
+    return 0;
+}
+
+/*
+ * Below are the only things that can work
+ */
+
+int _mkdir(const char* pathname) {
+    BOOL result = CreateDirectoryA(
+            pathname,
+            NULL
+    );
+
+    if (result == true) {
+        return 0;
+    } else {
+        DWORD err = GetLastError();
+
+        if (err == ERROR_ALREADY_EXISTS) {
```

Comment:
There's  alot of cases that are not covered from `GetLastError()`.  ERROR_DISK_FULL is just an obvious example.

Although these will fail, errno will not be set. Not sure if thats a problem